onnxruntime/core/framework/session_state.cc

```diff
@@ -143,10 +143,19 @@ common::Status SessionState::AddInputNameToNodeInfoMapping(const std::string& in
       if (current_provider == new_provider) {
         entries.push_back(node_info);
       } else {
-        return ORT_MAKE_STATUS(ONNXRUNTIME, NOT_IMPLEMENTED,
-                               "Using an input in multiple nodes on different devices is not supported currently. Input:",
-                               input_name, " is used by node ", existing_entry.p_node->Name(), " (", current_provider,
-                               ") and node ", node_info.p_node->Name(), " (", new_provider, ").");
+        // TODO: replace this with some more common check to allow CUDA/TensorRT sharing input too
+        static const std::unordered_set<std::string> cpu_compatible_providers = {
+            kCpuExecutionProvider,
+            kMklDnnExecutionProvider,
+            kNGraphExecutionProvider,
+            kNupharExecutionProvider};
+
```

Comment:
Should kOpenVINOExecutionProvider be in this list as well?

Reply:
Not sure, I thought OpenVINO is for non-CPU device? Seems not related to this PR anyway.

---
In reply to: [313178412](https://github.com/microsoft/onnxruntime/pull/881#discussion_r313178412) [](ancestors = 313178412)